internal/app/singularity/instance_linux.go

```diff
@@ -122,19 +130,129 @@ func WriteInstancePidFile(name, pidFile string) error {
 	return nil
 }
 
+// instanceListOrError is a private function to retrieve named instances or fail
+// We wrap the error from instance.List to provide a more specific error message
+func instanceListOrError(name, instanceUser string) ([]*instance.File, error) {
```

Comment:
I probably wouldn't pull this out in a function at this point, but it's just a preference thing. Some thoughts:

- The order of arguments should match `instance.List` i.e. `instanceUser, name` so it's not confusing.
- Use `%w` to wrap the error, rather than `%v`
- Make sure the comment specifically says `fail if there are no instances`